src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,10 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value') {
+          if (node.value != value) {
+            node.value = '' + value;
+          }
```

Comment:
Ironically our code used to look a lot like this… https://github.com/facebook/react/blob/0.14-stable/src/renderers/dom/shared/DOMPropertyOperations.js#L187-L197

@jimfb - I think this is more fallout from your changes. Can you take a look and figure out a way forward?